File: lintcode/climbing-stairs.py
#!/usr/bin/env python
# encoding: utf-8

# Question: http://www.lintcode.com/en/problem/climbing-stairs/

class Solution:
    """
    @param n: An integer
    @return: An integer
    """
    def climbStairs(self, n):
        # write your code here
        if n <= 2:
            return n

        i = 1
        j = 2
        r = 0
        for z in xrange(2, n):
            r = i + j
            i = j
            j = r
        return r


if __name__ == '__main__':
    s = Solution()
    print s.climbStairs(3)
    print s.climbStairs(4)
    print s.climbStairs(5)

File: leetcode-2/climbing-stairs.cc
// Question: https://leetcode.com/problems/climbing-stairs/

class Solution {
public:
    int climbStairs(int n) {
        if (n == 1 || n == 2) {
            return n;
        }

        int i = 1, j = 2;
        for (int k = 3; k <= n; ++k) {
            int r = i+j;
            i = j;
            j = r;
        }
        return j;
    }
};
